DataFormats/HcalRecHit/interface/HcalRecHitSoA.h

```diff
@@ -0,0 +1,19 @@
+#ifndef DataFormats_HcalRecHit_HcalRecHitSoA_h
+#define DataFormats_HcalRecHit_HcalRecHitSoA_h
+
+#include "DataFormats/SoATemplate/interface/SoALayout.h"
+
+namespace hcal {
+
+  GENERATE_SOA_LAYOUT(HcalRecHitSoALayout,
+                      SOA_SCALAR(uint32_t, size),
+                      SOA_COLUMN(uint32_t, did),
+                      SOA_COLUMN(float, energy),
+                      SOA_COLUMN(float, chi2),
+                      SOA_COLUMN(float, energyM0),
+                      SOA_COLUMN(float, timeM0))
```

Comment:
@jsamudio @hatakeyamak is this SoA suitable to be used as input for the HCAL PFRecHits producer ?

Reply:
To me this looks okay, although I could use some clarification of the difference between `energy` and `energyM0`

Reply:
From the name, I'd guess that `energyM0` should be the energy from "method 0", the old approach without any fit.

Reply:
Looks all right to me. What we really need at the end from HCAL rechits are those in:
https://github.com/cms-sw/cmssw/blob/master/DataFormats/ParticleFlowReco/interface/CaloRecHitSoA.h
and the above are sufficient.

Reply:
Indeed, `energyM0` is the method 0 energy, which simply sums the energy in SOI.
Question for @hatakeyamak @jsamudio  , is the `flag` column needed in HCAL rechits? 
https://github.com/cms-sw/cmssw/blob/master/DataFormats/ParticleFlowReco/interface/CaloRecHitSoA.h#L17

Reply:
we kept it in case we have to use it for potential noise flagging etc at some point, but I believe we are not using it right now.

Reply:
Right, not needed for HCAL